cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -246,7 +248,8 @@ private void resetRegistry() {
     registry = registryFactory.create();
     for (Map.Entry<String, ? extends DatasetModule> entry : defaultModules.entrySet()) {
       LOG.info("Adding Default module: " + entry.getKey() + " " + this.toString());
-      add(entry.getKey(), entry.getValue(), true);
+      // default modules in system namespace
+      add(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, entry.getKey()), entry.getValue(), true);
```

Comment:
I don't have to construct the Id because I hard code it to construct with SYSTEM_NAMESPACE here. The default modules are injected in ``DatasetServiceModules`` with Strings. They are also used with String module names in ``DatasetTypeManager`` and some tests. It would be necessary to construct ``defaultModules`` with ``Id.Namespace``, everywhere, if the namespace in it can change. I presumed that default modules can stay in the SYSTEM_NAMESPACE, they wouldn't exist independently per namespace. Correct me if my assumption is wrong - if so, yes we'd have to have ``defaultModules`` be a map from id to module.